### PR TITLE
Update Go packages

### PR DESCRIPTION
**Pull request by soharab-ic**
## Description

Updated following go packages:

```
github.com/elastic/crd-ref-docs: v0.0.10 -> v0.0.12
github.com/go-git/go-git/v5: v5.11.0 -> v5.12.0
github.com/prometheus/client_golang: v1.18.0 -> v1.19.0
github.com/prometheus/common: v0.46.0 -> v0.53.0
github.com/stretchr/testify: v1.8.4 -> v1.9.0
go.opentelemetry.io/contrib/instrumentation/net/http/otelhttp: v0.48.0 -> v0.51.0
go.opentelemetry.io/contrib/propagators/autoprop: v0.48.0 -> v0.51.0
go.opentelemetry.io/otel: v1.23.1 -> v1.26.0
go.opentelemetry.io/otel/exporters/otlp/otlptrace: v1.23.1 -> v1.26.0
go.opentelemetry.io/otel/exporters/otlp/otlptrace/otlptracegrpc: v1.23.1 -> v1.26.0
go.opentelemetry.io/otel/sdk: v1.23.1 -> v1.26.0
go.opentelemetry.io/otel/trace: v1.23.1 -> v1.26.0
go.uber.org/zap: v1.26.0 -> v1.27.0
golang.org/x/net: v0.21.0 -> v0.25.0
google.golang.org/grpc: v1.61.1 -> v1.63.2
k8s.io/api: v0.29.2 -> v0.30.0
k8s.io/apiextensions-apiserver: v0.29.2 -> v0.30.0
k8s.io/apimachinery: v0.29.2 -> v0.30.0
k8s.io/client-go: v0.29.2 -> v0.30.0
k8s.io/code-generator: v0.29.2 -> v0.30.0
k8s.io/metrics: v0.29.2 -> v0.30.0
sigs.k8s.io/controller-runtime: v0.17.1 -> v0.18.2
sigs.k8s.io/controller-tools: v0.14.0 -> v0.15.0
```

## Which issue(s) this PR fixes:

Fixes #

## Checklist:

- [ ] I ran tests as well as code linting locally to verify my changes. 
- [ ] I have done manual verification of my changes, changes working as expected.
- [ ] I have added new tests to cover my changes.
- [ ] My changes follow contributing guidelines of Fission.
- [x] I have signed all of my commits.

### CODE REVIEW

File: go.mod [hunk 1]
```diff
@@ -25,32 +25,32 @@ require (
...
+	golang.org/x/net v0.25.0
+	google.golang.org/grpc v1.63.2
+	k8s.io/api v0.30.0
+	k8s.io/apiextensions-apiserver v0.30.0
+	k8s.io/apimachinery v0.30.0
+	k8s.io/client-go v0.30.0
+	k8s.io/code-generator v0.30.0
+	k8s.io/metrics v0.30.0
+	sigs.k8s.io/controller-runtime v0.18.2
+	sigs.k8s.io/controller-tools v0.15.0
```
Comment (sanketsudake) [inline go.mod:53]: Update here also https://github.com/fission/fission/blob/241983d2f7873eca18e91fa80218eb0025f45a31/Makefile#L68

File: go.mod [hunk 2]
```diff
@@ -8,10 +8,10 @@ require (
 	github.com/dchest/uniuri v1.2.0
 	github.com/docopt/docopt-go v0.0.0-20180111231733-ee0de3bc6815
 	github.com/dustin/go-humanize v1.0.1
-	github.com/elastic/crd-ref-docs v0.0.10
+	github.com/elastic/crd-ref-docs v0.0.12
```
Comment (sanketsudake) [inline go.mod:11]: Update here also https://github.com/fission/fission/blob/241983d2f7873eca18e91fa80218eb0025f45a31/Makefile#L103

Comment (sanketsudake) [reply go.mod:11]: Please ensure you run `make all-generators` command and push all changed files along with these modules update.

**Review by sanketsudake — CHANGES_REQUESTED** [review]

File: hack/update-codegen.sh [hunk 3]
```diff
@@ -23,7 +23,7 @@ source "${CODEGEN_PKG}/kube_codegen.sh"
 kube::codegen::gen_client \
```
Comment (sanketsudake) [inline hack/update-codegen.sh:23]: Just change this to 
```
kube::codegen::gen_client \
	"./pkg/apis" \
	--with-watch \
	--with-applyconfig \
	--output-pkg github.com/fission/fission/pkg/generated \
	--output-dir "${OUTDIR}" \
	--boilerplate "${SCRIPT_ROOT}/hack/boilerplate.txt"
```

File: Makefile [hunk 4]
```diff
@@ -65,7 +65,7 @@ codegen: controller-gen-install
```
Comment (sanketsudake) [inline Makefile:65]: Also under codegen section need to change order

```
### Codegen
codegen: controller-gen-install
	@./hack/update-codegen.sh
	$(GOPATH)/bin/controller-gen object:headerFile="hack/boilerplate.txt" paths="./..."

```

File: go.mod (hunk 1 above)
Comment (soharab-ic) [reply go.mod:53]: Done

File: go.mod (hunk 2 above)
Comment (soharab-ic) [reply go.mod:11]: Done

File: go.mod [hunk 5]
```diff
@@ -25,32 +25,32 @@ require (
...
+	go.uber.org/zap v1.27.0
+	golang.org/x/net v0.25.0
+	google.golang.org/grpc v1.63.2
+	k8s.io/api v0.30.0
+	k8s.io/apiextensions-apiserver v0.30.0
+	k8s.io/apimachinery v0.30.0
+	k8s.io/client-go v0.30.0
+	k8s.io/code-generator v0.30.0
+	k8s.io/metrics v0.30.0
+	sigs.k8s.io/controller-runtime v0.18.2
```
[the remark this answers is not in the document]
Comment (soharab-ic) [reply go.mod:52]: Using `latest` tag only.

File: hack/update-codegen.sh (hunk 3 above)
Comment (soharab-ic) [reply hack/update-codegen.sh:23]: Done

File: Makefile (hunk 4 above)
Comment (soharab-ic) [reply Makefile:65]: Done

**Review by sanketsudake — APPROVED** [review]

**Pull request merged**